cmd/zoekt-webserver/main.go

```diff
@@ -322,15 +331,30 @@ func addProxyHandler(mux *http.ServeMux, socket string) {
 	mux.Handle("/indexserver/", http.StripPrefix("/indexserver/", http.HandlerFunc(proxy.ServeHTTP)))
 }
 
-// shutdownOnSignal will listen for SIGINT or SIGTERM and call
-// srv.Shutdown. Note it doesn't call anything else for shutting down. Notably
-// our RPC framework doesn't allow us to drain connections, so it when
-// Shutdown is called all inflight RPC requests will be closed.
-func shutdownOnSignal(srv *http.Server) error {
+// shutdownSignalChan returns a channel which is listening for shutdown
+// signals from the operating system.
+func shutdownSignalChan() <-chan os.Signal {
 	c := make(chan os.Signal, 3)
 	signal.Notify(c, os.Interrupt)    // terminal C-c and goreman
 	signal.Notify(c, syscall.SIGTERM) // Kubernetes
+	return c
+}
 
+// closeOnSignal will listen for SIGINT or SIGTERM and call srv.Close. This is
+// not a graceful shutdown, see shutdownOnSignal.
+func closeOnSignal(srv *http.Server) error {
+	c := shutdownSignalChan()
+	<-c
+
+	return srv.Close()
+}
+
+// shutdownOnSignal will listen for SIGINT or SIGTERM and call srv.Shutdown.
+// Note it doesn't call anything else for shutting down. Notably our RPC
+// framework doesn't allow us to drain connections, so when Shutdown we will
+// wait 10s before closing.
```

Comment:
Is this comment correct given that we don't call shutdownOnSignal if we start the server with -rpc?

Reply:
it is correct in for this function, but I will add a note explaining that we have that behaviour.